robots/assembly/base.rb

```diff
@@ -0,0 +1,31 @@
+module Robots
+  module DorRepo
+    module Assembly
+      class Base
+        include LyberCore::Robot
+
+        def workflow_service
+          Dor::Config.workflow.client
+        end
+
+        protected
+
+        def with_item(druid, items_only: Dor::Config.configure.assembly.items_only, &block)
+          ai = item(druid)
+
+          if (items_only && !ai.is_item?)
+             logger.warn("Skipping exif-collect for #{druid} since it is not an item")
```

Comment:
this warning message is hardcoded for `exif-collect` but as I read it, `jp2-create` and `checksum-compute` need to be handled here, right? maybe use the class name for the warning message?